fog/ingest/server/src/controller.rs

```diff
@@ -379,36 +379,39 @@ where
             // if we are still needed to be active.
             if state.is_active() {
                 log::trace!(self.logger, "publish report");
-                match self.publish_report(&ingress_pubkey, &mut state) {
-                    Ok(ingress_key_status) => {
-                        // If our key is retired, and the index we want to scan is past expiry,
-                        // early return. Note, we don't even NEED to scan
-                        // when block.index == pubkey_expiry, because the
-                        // semantic of pubkey expiry is that it bounds the tombstone block, and a
-                        // transaction cannot land in its tombstone block.
-                        // But scanning the tombstone block as well may help
-                        // deal with off-by-one errors somewhere else, and doesn't really hurt.
-                        if ingress_key_status.retired
-                            && block.index > ingress_key_status.pubkey_expiry
-                        {
-                            log::warn!(self.logger, "When preparing to process block index {}, we discovered that our ingress key is expired: {:?}. Switching to idle and nuking keys.", block.index, ingress_key_status);
-                            state.set_idle();
-                            self.new_egress_key(&mut state)
-                                .expect("Failure to rotate egress key can't be recovered from");
-                            return;
+                let mut retry_seconds = 1;
+                loop {
+                    match self.publish_report(&ingress_pubkey, &mut state) {
+                        Ok(ingress_key_status) => {
+                            // If our key is retired, and the index we want to scan is past expiry,
+                            // early return. Note, we don't even NEED to scan
+                            // when block.index == pubkey_expiry, because the
+                            // semantic of pubkey expiry is that it bounds the tombstone block, and
+                            // a transaction cannot land in its
+                            // tombstone block. But scanning the
```

Comment:
Nit*: Is this formatting intentional? It's weird how the column got so short suddenly.

Reply:
not really, there is some kind of "reflow comments" option that is turned on in rustfmt toml.

i think james turned it on, if you want to turn it off we can talk to him

Reply:
i can try to make this comment a little nicer